src/sentry/api/permissions.py

```diff
@@ -276,6 +227,56 @@ def determine_access(
                 raise MemberDisabledOverLimit(organization)
 
 
+class StaffPermissionMixin(SentryPermission):
```

Comment:
the class was moved down so I could change this line to inherit from `SentryPermission` -- the mixin is still somewhat unsound but at least it can enforce signatures correctly (and validate `super()`) after this